new/analyse/create-dependency-tree.ts

```diff
@@ -0,0 +1,62 @@
+import { Module } from '../types';
+import { clone } from 'ramda';
+import { yellow } from 'chalk';
+
+import { logger } from '../helpers/logger';
+
+const { error, log } = logger('[analyse] create dependency tree');
+
+export const createDependencyTree: Module = (env) => {
+  if (!env.packages) {
+    const errorMessage = 'no packages found on env config object';
+    error(errorMessage);
+    throw new Error(errorMessage);
+  }
+
+  const packagesClone = clone(env.packages);
+  const counter = {
+    dependencies: 0,
+    devDependencies: 0,
+    peerDependencies: 0,
+  };
+
+  log('Analysing dependency tree');
+
+  Object.keys(packagesClone).forEach((packageName) => {
+    counter.dependencies +=
+      packagesClone[packageName].dependsOn.dependencies.length;
+    packagesClone[packageName].dependsOn.dependencies.forEach((dependency) => {
+      packagesClone[dependency].dependingOnThis.dependencies.push(packageName);
+    });
+
+    counter.devDependencies +=
+      packagesClone[packageName].dependsOn.devDependencies.length;
+    packagesClone[packageName].dependsOn.devDependencies.forEach(
+      (dependency) => {
+        packagesClone[dependency].dependingOnThis.devDependencies.push(
+          packageName
```

Comment:
You only push the package name. We really need the current **range** here because, in one of the next steps, we analyse that.

We should end up with a structure like:

In dependsOn:
```js
{
  packageName: '@xingternal/button',
  dependencyPackageRange: '^3.1.1',
  type: 'devDependencies' // type can be also in the tree structure of the object as you did, but code gets simpler when it's a flat array
}
```

and on the other side:
```js
{
  packageName: '@xingternal/incremental-search',
  ownPackageRange: '^3.1.1', // the range of button inside incremental search!!
  type: 'devDependencies'
}
```

Reply:
As we agreed, I was planning to refactor this when needed in a future step

Reply:
👍 
(But I'd do it in a separate step before the next module)